### PR TITLE
Suggest using Object's convenience hashing functions for overriding hashCode

### LINKED ISSUE #3359: [2.14 Stable Release] Suggest using `Object` convenience hashing functions
Rolling your own hashing logic can potentially cause issues, so perhaps it would be a good idea for us to suggest the usage of the newly provided hash functions from https://github.com/dart-lang/sdk/commit/83376bf1ee81607a182f8558b242f1c9f4883246

This can help avoid potential implementation issues and consistency is always great.

https://github.com/dart-lang/sdk/blob/master/CHANGELOG.md#dartcore

### PR DESCRIPTION
**Pull request by parlough**
Fixes #3359

### CODE REVIEW

File: src/_guides/libraries/library-tour.md [hunk 1]
```diff
@@ -1511,6 +1519,12 @@ var random = Random();
 random.nextBool(); // true or false
 ```
 
+{{site.alert.warning}}
```
Comment (kevmoo) [inline src/_guides/libraries/library-tour.md:1522]: Uh...avoid unrelated changeds in a PR. Split this out.

File: src/_guides/libraries/library-tour.md [hunk 2]
```diff
@@ -919,6 +919,20 @@ also want to override the `==` operator. Objects that are equal (via
 `==`) must have identical hash codes. A hash code doesn’t have to be
 unique, but it should be well distributed.
 
+{{site.alert.tip}}
```
Comment (kevmoo) [inline src/_guides/libraries/library-tour.md:922]: Good idea!

File: src/_guides/libraries/library-tour.md (hunk 1 above)
Comment (parlough) [reply src/_guides/libraries/library-tour.md]: Woops, included this by accident, thanks for catching that! You can find those changes in https://github.com/dart-lang/site-www/pull/3664 :)

File: src/_guides/libraries/library-tour.md [hunk 3]
```diff
@@ -919,6 +919,20 @@ also want to override the `==` operator. Objects that are equal (via
...
+  To consistently and easily implement the `hashCode` getter,
+  consider using the static hashing methods provided by the `Object` class.
+
+  For multiple objects, consider using [`Object.hash`][], 
+  while for collections
+  consider [`Object.hashAll`][] or [`Object.hashAllUnordered`][] 
+  according to your object's definition of equality.
```
Comment (kwalrath) [inline src/_guides/libraries/library-tour.md:929]: I believe we use `()` after function names, so I added those. (They'd need to be added to the link definitions as well, of course.)

Also, I didn't understand the multiple objects vs. collections thing, and I wanted to reduce sentence length. So... Here's my attempt:

```suggestion
  To generate a single hash code for multiple objects,
  you can use [`Object.hash()`][].
  To generate a hash code for a collection,
  you can use either [`Object.hashAll()`][] (if element order matters)
  or [`Object.hashAllUnordered()`][].
```

I vaguely remember us talking about collection member terminology recently (_element? item? member?_) but I don't remember what we decided.

**Review by kwalrath — COMMENTED** [review]
Nice! I'd like to tweak the wording, but the code and gist of this lgtm.

File: src/_guides/libraries/library-tour.md (hunk 3 above)
Comment (parlough) [reply src/_guides/libraries/library-tour.md:929]: These changes look good to me, thanks. I do believe we discussed element as the best choice here and it makes sense in this case as well.

**Comment by parlough** [comment]
@lrhn Do you have any thoughts on this mention?

File: src/_guides/libraries/library-tour.md [hunk 4]
```diff
@@ -919,6 +919,21 @@ also want to override the `==` operator. Objects that are equal (via
 `==`) must have identical hash codes. A hash code doesn’t have to be
 unique, but it should be well distributed.
 
+{{site.alert.tip}}
+  To consistently and easily implement the `hashCode` getter,
+  consider using the static hashing methods provided by the `Object` class.
+
+  To generate a single hash code for multiple objects,
```
Comment (lrhn) [inline src/_guides/libraries/library-tour.md:926]: I'd say: "objects" -> "properties of an object"

File: src/_guides/libraries/library-tour.md [hunk 5]
```diff
@@ -933,15 +948,9 @@ class Person {
 
   Person(this.firstName, this.lastName);
 
-  // Override hashCode using strategy from Effective Java,
-  // Chapter 11.
+  // Override hashCode using Object's static hashing functions.
```
Comment (lrhn) [inline src/_guides/libraries/library-tour.md:951]: Object -> \`Object\`

**Review by lrhn — APPROVED** [review]

**Review by kwalrath — APPROVED** [review]
I don't think single quotes are really necessary for `//` comments, but they're not hurting anyone.

**Pull request merged**